http-client-spi/src/test/java/software/amazon/awssdk/http/SdkHttpRequestResponseTest.java

```diff
@@ -401,13 +402,41 @@ public BuilderProxy setMap(Map<String, List<String>> map) {
         });
     }
 
+    @Test
+    public void testSdkHttpFullRequestBuilderNoQueryParams() throws URISyntaxException {
+        URI uri = new URI("https://github.com/aws/aws-sdk-java-v2/issues/2034");
+        final SdkHttpFullRequest sdkHttpFullRequest = SdkHttpFullRequest.builder().method(SdkHttpMethod.POST).uri(uri).build();
+        assertThat(sdkHttpFullRequest.getUri().getQuery()).isNullOrEmpty();
+    }
+
+    @Test
+    public void testSdkHttpFullRequestBuilderDoNotKeepUriRequestParameters() throws URISyntaxException {
+        URI uri = new URI("https://github.com/aws/aws-sdk-java-v2/issues/2034?reqParam=1234&oParam=3456&reqParam=5678");
+        final SdkHttpFullRequest sdkHttpFullRequest = SdkHttpFullRequest.builder().method(SdkHttpMethod.POST).uri(uri).build();
+        assertThat(sdkHttpFullRequest.getUri().getQuery()).isNullOrEmpty();
+    }
+
+    @Test
+    public void testSdkHttpFullRequestBuilderUriWithQueryParams() throws URISyntaxException {
+        URI uri = new URI("https://github.com/aws/aws-sdk-java-v2/issues/2034?reqParam=1234&oParam=3456%26reqParam%3D5678");
+        final SdkHttpFullRequest sdkHttpFullRequest =
+            SdkHttpFullRequest.builder().method(SdkHttpMethod.POST).uriWithQueryParams(uri).build();
+        assertThat(sdkHttpFullRequest.getUri().getQuery()).contains("reqParam=1234", "oParam=3456", "reqParam=5678");
```

Comment:
I don't think this assertion is correct per my comment earlier re: parsing.

Splitting `reqParam=1234&oParam=3456%26reqParam%3D5678` on the `&` delimiter should give `reqParam=1234` and `oParam=3456%26reqParam%3D5678` as the two name-value pairs, then parsing each as a pair gives [reqParam, 1234] and `[oParam, 3456%26reqParam%3D5678]`

Reply:
Today I learned! updated and adapted unit test